kotlinx-coroutines-core/common/src/Delay.kt

```diff
@@ -95,7 +95,6 @@ public interface Delay {
  * }
  * ```
  */
-@ExperimentalCoroutinesApi
 public suspend fun awaitCancellation(): Nothing = suspendCancellableCoroutine {}
```

Comment:
I feel uneasy about this one.

On the one hand, it's okay the way it is. On the other hand, we have `awaitClose` that receives `block` to invoke on close (a bit inconsistent) and `hang` function in our tests that I use rather extensively.

WDYT about changing it to
```
public suspend [inline] fun awaitCancellation(afterCancellation: () -> Unit = {}): Nothing {
    try {
        suspendCancellableCoroutine<Nothing> {}
    } finally {
        afterCancellation()
    }
}
```
?

Reply:
When I submitted `awaitCancellation()`, @elizarov told me the bytecode seemed to be more efficient without the `inline` keyword, so I dropped it (though I don't know how to check and compare these matters yet).

Assuming this is right, shouldn't we have an overload of `awaitCancellation` that takes no default value for its lambda parameter?

Reply:
I'd vote for stabilizing `awaitCancellation()` as is. If we find real-life use-cases for adding `awaitClose`-like block to it, then we can always add the corresponding inline overload later in the future. Moreover, it would be more efficient to implement a variant with a block as an inline function that is calling this non-inline `awaitCancellation()` without a block, wrapping it into `try/finally` (`suspendCancellableCoroutine<Nothing> {}` produces much more bytecode).